src/auro-badge.js

```diff
@@ -24,18 +25,35 @@ import styleCss from "./style-css.js";
 
 // build the component class
 class AuroBadge extends LitElement {
-  // constructor() {
-  //   super();
-  // }
+  constructor() {
+    super();
+    this.dom = new DOMParser().parseFromString(closeIcon.svg, 'text/html');
+    this.svg = this.dom.body.firstChild;
+  }
 
   // function to define props used within the scope of this component
   static get properties() {
     return {
       // ...super.properties,
-      cssClass:   { type: String }
+      action: {type: Boolean},
     };
   }
 
+  /**
+   * Fires a custom event and removes the element from the DOM if action is true
+   *
+   * @param {*} event interaction event from Badge
+   * @returns {void}
+   */
+  handleChange(event) {
+    if (this.action) {
+      const customEvent = new CustomEvent(event.type, event);
+
+      this.dispatchEvent(customEvent);
+      this.remove();
```

Comment:
What will happen to the user's keyboard focus when the element is removed from the DOM?

Reply:
The intended functionality, I presume, is for it to return focus to the closest-to-left sibling element before it's removed - good point, I'll add this in.

Reply:
I can't say that I ever really paid attention to this and I can't really find any good examples on the open web. 

I think the question here is, what do we 'want' to happen here? Moving to the left sibling without an announcement is not really helpful. The user has no way of know that this happened. 

Having said that, I also __don't feel that this is the concern of the `badge` element__. The badge by itself is responsible for a visual attribute and the placement of an action versus just content. It's a11y role in a larger context is outside the scope of this custom element. 

Now if we made a larger element that contained a search feature and the badges, then that would be the proper context to enforce a11y.

Reply:
Good points -- I agree that this is outside of the scope of the component. The consumer of this component will need to manage focus appropriately by listening to the click event, since they understand the surrounding context.

Reply:
@gusnaughton ^^ for you. So, the result here is that you do not need to be concerned about the question that Geoff posed here. 

Move on and profit!